language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/semantictokens/SemanticTokensVisitor.java

```diff
@@ -601,14 +615,19 @@ private void processSymbols(Node node, LinePosition startLine) {
             final int type = referenceType;
             final int modifiers = referenceModifiers == -1 ? 0 : referenceModifiers;
 
+            // No need of finding reference for strings since those are covered from the variable declaration.
+            if (node.kind() == SyntaxKind.STRING_LITERAL) {
```

Comment:
Shouldn't we do this at the top of the method? Check whether when the node is a string literal it has an impact within the other if clauses

Reply:
+1. Updated